source/material_model/interface.cc

```diff
@@ -370,9 +370,17 @@ namespace aspect
       fe_values[introspection.extractors.pressure].get_function_values (solution_vector, this->pressure);
       fe_values[introspection.extractors.pressure].get_function_gradients (solution_vector, this->pressure_gradient);
       if (use_strain_rate)
+      {
         fe_values[introspection.extractors.velocities].get_function_symmetric_gradients (solution_vector,this->strain_rate);
+        requested_properties = requested_properties | MaterialProperties::viscosity;
+      }
       else
+      {
         this->strain_rate.resize(0);
+        requested_properties = MaterialProperties::equation_of_state_properties |
+                                         MaterialProperties::reaction_terms |
+                                         MaterialProperties::additional_outputs;
```

Comment:
This way you overwrite the existing value of requested_properties, which is different from the behavior in the if-branch. I think you were looking for only disabling viscosity? If so that would probably work like this:
```suggestion
        requested_properties = requested_properties & ~MaterialProperties::viscosity;
```

Reply:
Ah didn't know about `~`, cool!